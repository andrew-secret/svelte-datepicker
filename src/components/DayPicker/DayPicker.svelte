<script lang="ts">
  import type { IUtils } from "@date-io/core/IUtils";
  export let dateAdapter: IUtils<Date>;
  export let fullMonth: Date[][];
  export let selectedDates: Date[];
  export let currentMonthNumber: number;
  export let focusedDay: Date;
  export let minDate: Date;
  export let maxDate: Date;
  export let onDaySelect: (day: Date) => void;
  export let handleFocus: (focusedDay: Date) => void;

  let currentDay = dateAdapter.date() as Date;

  function handleDisabledDay(day: Date): boolean {
    const isDayDisabled =
      !dateAdapter.isWithinRange(day, [minDate, maxDate]) &&
      !dateAdapter.isSameDay(day, minDate) &&
      !dateAdapter.isSameDay(day, maxDate);

    return isDayDisabled;
  }

  function handleTabIndex(day: Date): number {
    const isSelected = selectedDates.some(
      (selectedDate) => selectedDate && dateAdapter.isSameDay(selectedDate, day)
    );

    const isFocused = dateAdapter.isSameDay(day, focusedDay);

    if (isSelected || isFocused) {
      return 0;
    }
    return -1;
  }

</script>

<div class="weekdays">
  {#each dateAdapter.getWeekdays() as day}
    <span class="weekday-label" title={day}>
      {day}
    </span>
  {/each}
</div>

<div role="grid">
  {#each fullMonth as week}
    <div class="week" role="row">
      {#each week as day, i}
        <!-- svelte-ignore a11y-autofocus  -->
        <div role="cell">
          <button
            aria-hidden={dateAdapter.getMonth(day) !== currentMonthNumber}
            aria-label={day}
            type="button"
            class="day"
            disabled={handleDisabledDay(day)}
            data-testid={dateAdapter.getMonth(day) !== currentMonthNumber
              ? "hidden-day"
              : "visible-day"}
            tabindex={handleTabIndex(day)}
            class:today={dateAdapter.isSameDay(day, currentDay)}
            class:selected={selectedDates.some(
              (selectedDate) =>
                selectedDate && dateAdapter.isSameDay(selectedDate, day)
            )}
            class:outside-of-current-month={dateAdapter.getMonth(day) !==
              currentMonthNumber}
            on:click={() => {
              onDaySelect(day);
            }}
            on:focus={() => handleFocus(day)}
            autofocus={focusedDay !== null &&
              dateAdapter.isSameDay(day, focusedDay)}
          >
            {dateAdapter.format(day, "dayOfMonth")}
          </button>
        </div>
      {/each}
    </div>
  {/each}
</div>

<style lang="scss">
  .weekdays {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 42px;
  }

  div[role="grid"] {
    height: 280px;
  }

  .weekday-label {
    color: var(--sdp-color-grey-400);
    font-family: var(--sdp-font-family);
    width: 42px;
    height: 42px;
    margin: 0 4px;
    display: flex;
    text-align: center;
    align-items: center;
    justify-content: center;
  }

  .week {
    margin: 4px 0;
    display: flex;
    justify-content: center;
  }

  .day {
    color: var(--sdp-base-font-color);
    width: 42px;
    height: 42px;
    padding: 0;
    font-size: 0.85rem;
    font-family: var(--sdp-font-family);
    font-weight: 400;
    cursor: pointer;
    line-height: 1.66;
    border-radius: var(--sdp-border-radius);
    letter-spacing: 0.03333em;
    background-color: var(--sdp-btn-bg-color);
    transition: background-color 200ms ease-in-out;
    border: none;
    margin: 0 2px;

    &:hover {
      color: var(--sdp-hover-color);
      background-color: var(--sdp-hover-bg-color);
    }

    &:focus {
      outline: none;
      color: var(--sdp-focus-color);
      box-shadow: inset 0 0 0 0.15rem var(--sdp-bg-focus-color);
    }

    &:disabled {
      opacity: 0.35;
      pointer-events: none;
    }
  }

  .day:active {
    color: var(--sdp-active-color);
    background-color: var(--sdp-bg-active-color);
  }

  .day.selected {
    color: var(--sdp-selected-color);
    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
    font-weight: 500;
    background-color: var(--sdp-bg-selected-color);

    &:focus {
      outline: none;
      box-shadow: 0 0 0 0.15rem var(--sdp-bg-focus-color),
        inset 0 0 0 0.1rem var(--sdp-bg-color);
    }
  }

  .outside-of-current-month {
    visibility: hidden;
  }

  .today:not(.selected) {
    box-shadow: inset 0 0 0 0.15rem var(--sdp-base-font-color);

    &:focus {
      outline: none;
      color: var(--sdp-focus-color);
      box-shadow: inset 0 0 0 0.15rem var(--sdp-bg-focus-color);
    }
  }

</style>
