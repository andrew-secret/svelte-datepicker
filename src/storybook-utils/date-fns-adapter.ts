import LuxonAdapter from '@date-io/luxon';
import DateFnsAdapter from "@date-io/date-fns";

export const dateFnsUtils = new DateFnsAdapter();
export const luxonUtils = new LuxonAdapter();