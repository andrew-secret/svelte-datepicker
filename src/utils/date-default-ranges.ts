  export const defaultMinDate = new Date("1900-01-01");
  export const defaultMaxDate = new Date("2099-12-31");