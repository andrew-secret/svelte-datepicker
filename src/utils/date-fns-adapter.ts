import LuxonAdapter from '@date-io/luxon';
import DateFnsAdapter from "@date-io/date-fns";
import DayjsAdapter from '@date-io/dayjs'
import MomentUtils from '@date-io/moment'

// Should only be used in Storybook 
export const dateFnsUtils = new DateFnsAdapter();
export const luxonUtils = new LuxonAdapter();
export const dayJsUtils = new DayjsAdapter();
export const momentUtils = new MomentUtils();
