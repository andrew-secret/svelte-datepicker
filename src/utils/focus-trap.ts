export function handleFocusTrap(event: KeyboardEvent, {element, classNames, isFocusTrapDisabled}): void {
    if (!element || isFocusTrapDisabled) {
      return;
    }

    const focusableElements = element.querySelectorAll(classNames);
    const firstFocusableElement = focusableElements[0];
    const lastFocusableElement =
      focusableElements[focusableElements.length - 1];
    const isTabPressed = event.key === "Tab";

    if (!isTabPressed) {
      return;
    }

    /*
      the focusElements.length !== -1 check is needed to ensure that the 
      focusable elements array contains all elements. The reason for 
      this behavior is DOM manipulation due to the year picker.
    */
    if (
      !event.shiftKey &&
      document.activeElement === lastFocusableElement &&
      focusableElements.length !== 1
    ) {
      (firstFocusableElement as HTMLButtonElement).focus();
      event.preventDefault();
    }

    if (
      event.shiftKey &&
      document.activeElement === firstFocusableElement &&
      focusableElements.length !== 1
    ) {
      (lastFocusableElement as HTMLButtonElement).focus();
      event.preventDefault();
    }
  }