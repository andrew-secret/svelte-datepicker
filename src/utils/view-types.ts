export type view = "days" | "years";