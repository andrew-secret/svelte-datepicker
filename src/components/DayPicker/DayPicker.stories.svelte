<script lang="ts">
  import { Meta, Template, Story } from "@storybook/addon-svelte-csf";
  import {
    defaultMaxDate,
    defaultMinDate,
  } from "../../utils/date-default-ranges";
  import { dateFnsUtils } from "../../utils/date-fns-adapter";
  import DayPicker from "./DayPicker.svelte";

  let date = new Date("2021-01-04");
  let currentMonth = date;
</script>

<Meta title="Example/DayPicker" component={DayPicker} />

<Template>
  <DayPicker
    dateAdapter={dateFnsUtils}
    fullMonth={dateFnsUtils.getWeekArray(currentMonth)}
    currentMonthNumber={dateFnsUtils.getMonth(currentMonth)}
    selectedDates={[date]}
    minDate={defaultMinDate}
    maxDate={defaultMaxDate}
    handleFocus={() => null}
    onDaySelect={() => null}
  />
</Template>

<Story name="Default" />
