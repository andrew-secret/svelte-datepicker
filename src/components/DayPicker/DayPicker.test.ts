import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import { defaultMaxDate, defaultMinDate } from '../../utils/date-default-ranges';
import { dateFnsUtils } from '../../utils/date-fns-adapter';
import DayPicker from './DayPicker.svelte';

describe('DayPicker', () => {
    const date = new Date('2021-01-04');
    const currentMonth = date;
    const fullMonth = dateFnsUtils.getWeekArray(currentMonth);
    const currentMonthNumber = dateFnsUtils.getMonth(currentMonth);

    const props = {
        dateAdapter: dateFnsUtils,
        fullMonth,
        minDate: defaultMinDate,
        maxDate: defaultMaxDate,
        selectedDates: [date],
        focusedDay: date,
        currentMonthNumber,
        onDaySelect: jest.fn(),
        handleFocus: jest.fn(),
    }

    it('should render component', () => {
        const { container } = render(DayPicker, {
            props,
        });
        expect(container).toMatchSnapshot();
    });

    it('should not render days from previous or next month', () => {
        const { getAllByTestId } = render(DayPicker, {
            props,
        });
        const allHiddenDays = getAllByTestId('hidden-day');

        allHiddenDays.map(hiddenDay => {
            expect(hiddenDay).not.toBeVisible();
        });
    });
});