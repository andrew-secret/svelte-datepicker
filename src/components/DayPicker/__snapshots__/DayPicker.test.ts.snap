// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayPicker should render component 1`] = `
<body>
  <div>
    <div
      class="weekdays svelte-ie0jw4"
    >
      <span
        class="weekday-label svelte-ie0jw4"
        title="Su"
      >
        Su
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="Mo"
      >
        Mo
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="Tu"
      >
        Tu
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="We"
      >
        We
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="Th"
      >
        Th
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="Fr"
      >
        Fr
         
      </span>
      <span
        class="weekday-label svelte-ie0jw4"
        title="Sa"
      >
        Sa
         
      </span>
    </div>
     
    <div
      class="svelte-ie0jw4"
      role="grid"
    >
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Sun Dec 27 2020 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            27
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Mon Dec 28 2020 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            28
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Tue Dec 29 2020 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            29
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Wed Dec 30 2020 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            30
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Thu Dec 31 2020 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            31
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Fri Jan 01 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            1
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sat Jan 02 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            2
          </button>
        </div>
         
      </div>
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sun Jan 03 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            3
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Mon Jan 04 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            autofocus=""
            class="day svelte-ie0jw4 selected"
            data-testid="visible-day"
            tabindex="0"
            type="button"
          >
            4
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Tue Jan 05 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            5
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Wed Jan 06 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            6
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Thu Jan 07 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            7
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Fri Jan 08 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            8
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sat Jan 09 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            9
          </button>
        </div>
         
      </div>
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sun Jan 10 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            10
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Mon Jan 11 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            11
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Tue Jan 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            12
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Wed Jan 13 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            13
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Thu Jan 14 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            14
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Fri Jan 15 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            15
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sat Jan 16 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            16
          </button>
        </div>
         
      </div>
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sun Jan 17 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            17
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Mon Jan 18 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            18
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Tue Jan 19 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            19
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Wed Jan 20 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            20
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Thu Jan 21 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            21
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Fri Jan 22 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            22
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sat Jan 23 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            23
          </button>
        </div>
         
      </div>
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sun Jan 24 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            24
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Mon Jan 25 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            25
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Tue Jan 26 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            26
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Wed Jan 27 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            27
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Thu Jan 28 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            28
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Fri Jan 29 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            29
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sat Jan 30 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            30
          </button>
        </div>
         
      </div>
      <div
        class="week svelte-ie0jw4"
        role="row"
      >
        <div
          role="cell"
        >
          <button
            aria-hidden="false"
            aria-label="Sun Jan 31 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4"
            data-testid="visible-day"
            tabindex="-1"
            type="button"
          >
            31
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Mon Feb 01 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            1
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Tue Feb 02 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            2
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Wed Feb 03 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            3
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Thu Feb 04 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            4
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Fri Feb 05 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            5
          </button>
        </div>
        <div
          role="cell"
        >
          <button
            aria-hidden="true"
            aria-label="Sat Feb 06 2021 00:00:00 GMT+0000 (Coordinated Universal Time)"
            class="day svelte-ie0jw4 outside-of-current-month"
            data-testid="hidden-day"
            tabindex="-1"
            type="button"
          >
            6
          </button>
        </div>
         
      </div>
    </div>
  </div>
</body>
`;
